src/Compilers/CSharp/Portable/Binder/ExpressionVariableFinder.cs

```diff
@@ -91,30 +79,20 @@ private void VisitNodeToBind(CSharpSyntaxNode node)
             _nodeToBind = previousNodeToBind;
         }
 
-        internal static void FindExpressionVariables(
-            Binder binder,
-            ArrayBuilder<LocalSymbol> builder,
+        protected void FindExpressionVariables(
+            ArrayBuilder<TFieldOrLocalSymbol> builder,
             SeparatedSyntaxList<ExpressionSyntax> nodes)
         {
-            if (nodes.Count == 0)
-            {
-                return;
-            }
-
-            var finder = s_poolInstance.Allocate();
-            finder._scopeBinder = binder;
-            finder._enclosingBinder = binder;
-            finder._localsBuilder = builder;
+            Debug.Assert(nodes.Count > 0);
+            ArrayBuilder<TFieldOrLocalSymbol> save = _localsBuilder;
+            _localsBuilder = builder;
 
             foreach (var n in nodes)
             {
-                finder.VisitNodeToBind(n);
+                VisitNodeToBind(n);
```

Comment:
These are ExpressionSyntax nodes, the purpose of the assert is to verify other kinds of nodes.